internal/tools/uvmboot/lcow.go

```diff
@@ -81,94 +105,163 @@ var lcowCommand = cli.Command{
-					}
-				}
-			}
-			if c.IsSet(consolePipeArgName) {
-				options.ConsolePipe = c.String(consolePipeArgName)
+			options, err := createLCOWOptions(ctx, c, id)
+			if err != nil {
+				return err
 			}
 
-			if err := runLCOW(context.TODO(), options, c); err != nil {
+			if err := runLCOW(ctx, options, c); err != nil {
 				return err
 			}
+
 			return nil
 		})
 
 		return nil
 	},
 }
 
+func createLCOWOptions(_ context.Context, c *cli.Context, id string) (*uvm.OptionsLCOW, error) {
+	options := uvm.NewDefaultOptionsLCOW(id, "")
+	setGlobalOptions(c, options.Options)
+
+	// boot
+	if c.IsSet(bootFilesPathArgName) {
+		options.BootFilesPath = c.String(bootFilesPathArgName)
+	}
+
+	// kernel
+	if c.IsSet(kernelDirectArgName) {
+		options.KernelDirect = c.Bool(kernelDirectArgName)
+	}
+	if c.IsSet(kernelFileArgName) {
+		switch strings.ToLower(c.String(kernelFileArgName)) {
+		case uvm.KernelFile:
+			options.KernelFile = uvm.KernelFile
+		case uvm.UncompressedKernelFile:
+			options.KernelFile = uvm.UncompressedKernelFile
+		default:
+			return nil, unrecognizedError(c.String(kernelFileArgName), kernelFileArgName)
+		}
+	}
+	if c.IsSet(kernelArgsArgName) {
+		options.KernelBootOptions = c.String(kernelArgsArgName)
+	}
+
+	// rootfs
+	if c.IsSet(rootFSTypeArgName) {
+		switch strings.ToLower(c.String(rootFSTypeArgName)) {
+		case "initrd":
+			options.RootFSFile = uvm.InitrdFile
+			options.PreferredRootFSType = uvm.PreferredRootFSTypeInitRd
+		case "vhd":
+			options.RootFSFile = uvm.VhdFile
+			options.PreferredRootFSType = uvm.PreferredRootFSTypeVHD
+		default:
+			return nil, unrecognizedError(c.String(rootFSTypeArgName), rootFSTypeArgName)
+		}
+	}
+
+	if c.IsSet(vpMemMaxCountArgName) {
+		options.VPMemDeviceCount = uint32(c.Uint(vpMemMaxCountArgName))
+	}
+	if c.IsSet(vpMemMaxSizeArgName) {
+		options.VPMemSizeBytes = c.Uint64(vpMemMaxSizeArgName) * 1024 * 1024 // convert from MB to bytes
+	}
+
+	// GCS
+	options.UseGuestConnection = useGCS
+	if !useGCS {
+		if c.IsSet(execCommandLineArgName) {
+			options.ExecCommandLine = c.String(execCommandLineArgName)
+		}
+		if c.IsSet(forwardStdoutArgName) {
+			options.ForwardStdout = c.Bool(forwardStdoutArgName)
+		}
+		if c.IsSet(forwardStderrArgName) {
+			options.ForwardStderr = c.Bool(forwardStderrArgName)
+		}
+		if c.IsSet(outputHandlingArgName) {
+			switch strings.ToLower(c.String(outputHandlingArgName)) {
+			case "stdout":
+				options.OutputHandler = uvm.OutputHandler(func(r io.Reader) {
+					_, _ = io.Copy(os.Stdout, r)
+				})
+			default:
+				return nil, unrecognizedError(c.String(outputHandlingArgName), outputHandlingArgName)
+			}
+		}
+	}
+	if c.IsSet(consolePipeArgName) {
+		options.ConsolePipe = c.String(consolePipeArgName)
+	}
+
+	// general settings
+	if lcowDisableTimeSync {
+		options.DisableTimeSyncService = true
+	}
+
+	if c.IsSet(securityPolicyArgName) {
+		options.SecurityPolicy = c.String(options.SecurityPolicy)
```

Comment:
I started making similar changes: https://github.com/anmaxvl/hcsshim/commit/d1c5f4a307664237830649888c077c34570b32f2, but didn't get a chance to actually finalize it. We also need to expose `NoSecurityHardware` option through a flag to run on non-secure hardware.

Reply:
Does this align with your implementation? I'd rather you add the security hardware nuances, so ill adjust accordingly

Reply:
sounds good. leave the secure hardware part to me then. I think there's also `options.SecurityPolicyEnabled = true` missing.